WoWsShipBuilder.Common/Infrastructure/DataTransfer/AppNavigator.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System.ComponentModel;
+using Microsoft.AspNetCore.Components;
+using MudBlazor;
+using WoWsShipBuilder.Infrastructure.Metrics;
+
+namespace WoWsShipBuilder.Infrastructure.DataTransfer;
+
+public class AppNavigator
+{
+    private const int TransferLimit = 250;
+
+    private const string ShipStatsBaseUrl = "ship?shipIndexes=";
+    private const string BallisticChartsBaseUrl = "charts?shipIndex=";
+    private const string AccelerationChartsBaseUrl = "acceleration-charts?shipIndexes=";
+    private const string ShipComparisonBaseUrl = "comparison?p=";
+
+    private readonly SessionStateCache sessionStateCache;
+    private readonly NavigationManager navManager;
+    private readonly MetricsService metricsService;
+    private readonly ISnackbar snackbar;
+
+    public AppNavigator(SessionStateCache sessionStateCache, NavigationManager navManager, MetricsService metricsService, ISnackbar snackbar)
+    {
+        this.sessionStateCache = sessionStateCache;
+        this.navManager = navManager;
+        this.metricsService = metricsService;
+        this.snackbar = snackbar;
+    }
+
+    public enum AppPage
+    {
+        ShipStats,
+        BallisticCharts,
+        AccelerationCharts,
+        ShipComparison,
+    }
+
+    private static string GenerateDestinationUrl(AppPage destinationPage, IEnumerable<ShipBuildContainer> containerList, string? shellIndex)
```

Comment:
The method (and member) ordering in this class is still wrong, see my comment from the last review